src/nvim/ui.h

```diff
@@ -22,6 +23,7 @@ struct ui_t {
   void (*clear)(UI *ui);
   void (*eol_clear)(UI *ui);
   void (*cursor_goto)(UI *ui, int row, int col);
+  void (*cursor_style_set)(UI *ui, Dictionary args); // or a Dictionary ?
```

Comment:
You can store the entire cursor style in a Dictionary (which may have nested Dictionaries).